napari/utils/colormaps/colormap.py

```diff
@@ -827,10 +832,15 @@ def _labels_raw_to_texture_direct_numpy(
 
     See `_cast_labels_data_to_texture_dtype_direct` for more details.
     """
+    if direct_colormap.use_selection:
+        return (data == direct_colormap.selection).astype(np.uint8)
     mapper = direct_colormap._array_map
-
-    if data.dtype.itemsize > 2:
+    if any(x < 0 for x in direct_colormap.color_dict if x is not None):
```

Comment:
Do we need to add a test for this condition? I'm also confused about why the itemsize is no longer checked. Is it because the else clause is a no-op for (u)int(8|16)?

Reply:
added